backend/core/src/listings/listings.service.ts

```diff
@@ -170,7 +170,7 @@ export class ListingsService {
       if (!unit.id) {
         delete unit.id
       }
-      if (unit.status === "available") {
+      if (listingDto.listingAvailability === ListingAvailability.availableUnits) {
```

Comment:
Can we just check the units length once, rather than increment for every unit?

Reply:
good call, updated